src/chunk.c

```diff
@@ -880,6 +989,190 @@ chunk_find_all_oids(Hyperspace *hs, List *dimension_vecs, LOCKMODE lockmode)
 	return oid_list;
 }
 
+Datum
+chunk_show(PG_FUNCTION_ARGS)
+{
+	FuncCallContext *funcctx;
+	int			call_cntr;
+	int			max_calls;
+	TupleDesc	tupdesc;
+	AttInMetadata *attinmeta;
+
+	List	   *result_set = NIL;
+
+	/* stuff done only on the first call of the function */
+	if (SRF_IS_FIRSTCALL())
+	{
+		Cache	   *hypertable_cache;
+		Hypertable *ht;
+		Dimension  *time_dim;
+
+		/*
+		 * contains the list of hypertables chuns of which need to be
+		 * considred. this is a single hypertable if PG_ARGISNULL(0) is false
+		 * and is a valid hypertable Otherwise, it will have the list of all
+		 * hypertables in the system
+		 */
+		List	   *hypertables = NIL;
+		DimensionVec *slices;
+
+		ListCell   *lc;
+		Oid			time_dim_type = InvalidOid;
+
+
+		Oid			table_relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+
+		/*
+		 * get_fn_expr_argtype defaults to UNKNOWNOID if NULL but making it
+		 * InvalidOid makes the logic simpler later
+		 */
+		Oid			older_than_type = PG_ARGISNULL(1) ? InvalidOid : get_fn_expr_argtype(fcinfo->flinfo, 1);
+		Oid			newer_than_type = PG_ARGISNULL(2) ? InvalidOid : get_fn_expr_argtype(fcinfo->flinfo, 2);
+		int64		older_than = -1;
+		int64		newer_than = -1;
+		StrategyNumber start_strategy = InvalidStrategy;
+		StrategyNumber end_strategy = InvalidStrategy;
+
+		if (older_than_type != InvalidOid &&
+			newer_than_type != InvalidOid &&
+			older_than_type != newer_than_type)
+			/* Q:: maybe give current type information? */
```

Comment:
I can easily give the oid of the type. But do we want it in a human readable form?

Reply:
the current error msg is fine